### ISSUE TITLE
Unable to infer base url with Spring 5 and Boot 2

### ISSUE THREAD

**Issue by mapto** [springfox/springfox#2710]
First of all, congratulations for your great work on such an important project! However, I have an issue that emerged with the current snapshot.

I have a project that worked with Springfox 2.9.2. I use springfox-swagger2 and springfox-swagger-ui only. I had to [shadow](http://imperceptiblethoughts.com/shadow/) these due to conflicting dependencies of guava versions. This worked fine.

Now, I've upgraded to Spring 5/Boot 2, as the new version does not depend on guava, and thus the shadowing is not necessary any more. I've updated to 3.0.0-SNAPSHOT (cloned what is the current [origin/HEAD](https://github.com/springfox/springfox/commit/8162c43ae3af054fc5be87e4820780f43ad83da7)). As a consequence, I've had to substitute @EnableSwagger2 with @EnableSwagger2WebMvc (I've seen that Webflux still has other issues). My project has to use maven, so I had to manually build the swagger dependency modules and inject them into my local repository.

Unfortunately, I cannot expose the Swagger UI now. My REST services are accessible, but swagger-ui.html shows the "Unable to infer base url" prompt. I am also unable to open the /v2/api-docs resource. This project does not employ any security.

My application is published at the localhost root, so the URL of the UI is http://localhost/swagger-ui.html. I expect to see the api-docs resource at http://localhost/v2/api-docs 

Here's my configuration:

    @Configuration
    @EnableSwagger2WebMvc
    public class SwaggerConfig {

      @Autowired
      private BuildProperties buildProperties;

      @Bean
      public Docket api() {
        return new Docket(DocumentationType.SWAGGER_2)
                .select()
                .apis(RequestHandlerSelectors.basePackage("it.unriddle.test.controller"))
                .paths(PathSelectors.any())
                .build()
                .pathMapping("/")
                .apiInfo(metaData());
      }

      private ApiInfo metaData() {
        return new ApiInfoBuilder()
                .title("Test REST API")
                .version(buildProperties.getVersion())
                .build();
      }
    }

Any ideas what might cause the problem? Has anyone been able to investigate a similar issues?

Thanks,
Martin

**Comment by fennekit** [comment]
Same issue here

**Comment by oeresundsgruppen** [comment]
me 2

**Comment by dilipkrish** [comment]
@mapto I've been in a months hiatus just to avoid burnout. I plan to work on this issue, since this is an opportunity to make this better as we are free to make breaking changes. Will defenitely chalk this issue up to address.

**Comment by BiLuoHen** [comment]
@mapto @fennekit @oeresundsgruppen add the dependency `springfox-spring-webmvc` will work fine

**Comment by oeresundsgruppen** [comment]
Got it working in Tomcat finally - example https://github.com/oeresundsgruppen/swaggerTest, it was missing dispatcher setup

**Comment by dilipkrish** [comment]
Thats great @oeresundsgruppen!

**Comment by mapto** [comment]
I also managed to resolve my issue. I confirm that it had to do with depenencies, not only springfox-spring-webmvc, but also others. Practically, because of issues with shadowing, I couldn't use a released version. 3.0.0-SNAPSHOT is not yet on Maven, so I had to manually include in my pom.xml all dependencies that would otherwise be managed by gradle. This was quite a lot of back-and-forth between exceptions (missing both Java classes and web resources) and identifying which dependency the correponding packages belong.
